### PR TITLE
[FB Internal] Upgrade tool chain

### PR DESCRIPTION
**Pull request by siying**
Summary: Upgrade tool chain to the latest. It is done mostly manually as build_tools/build_detect_platform fails to update many of them.

Try to fix a new clang analyze warning with the new tool chain.

Test Plan: "make all", "USE_CLANG=1 make all"

### CODE REVIEW

**Comment by riversand963** [comment]
Can we also test `USE_CLANG=1 make analyze`?

**Comment by siying** [comment]
> Can we also test `USE_CLANG=1 make analyze`?

Sure let me do it now.

File: build_tools/fbcode_config_platform009.sh [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+DEPS_INCLUDE="$SNAPPY_INCLUDE $ZLIB_INCLUDE $BZIP_INCLUDE $LZ4_INCLUDE $ZSTD_INCLUDE $GFLAGS_INCLUDE $NUMA_INCLUDE $TBB_INCLUDE $LIBURING_INCLUDE"
+
+STDLIBS="-L $GCC_BASE/lib64"
+
+CLANG_BIN="$CLANG_BASE/bin"
+CLANG_LIB="$CLANG_BASE/lib"
+CLANG_SRC="$CLANG_BASE/../../src"
+
+CLANG_ANALYZER="$CLANG_BASE/bin/clang++"
```
Comment (riversand963) [inline build_tools/fbcode_config_platform009.sh:114]: `$CLANG_BIN/clang++`?

File: build_tools/fbcode_config_platform009.sh [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+  CFLAGS+=" -B$BINUTILS/gold -nostdinc -nostdlib"
+  CFLAGS+=" -isystem $LIBGCC_BASE/include/c++/9.x "
+  CFLAGS+=" -isystem $LIBGCC_BASE/include/c++/9.x/x86_64-facebook-linux "
+  CFLAGS+=" -isystem $GLIBC_INCLUDE"
+  CFLAGS+=" -isystem $LIBGCC_INCLUDE"
+  CFLAGS+=" -isystem $CLANG_INCLUDE"
+  CFLAGS+=" -isystem $KERNEL_HEADERS_INCLUDE/linux "
+  CFLAGS+=" -isystem $KERNEL_HEADERS_INCLUDE "
+  #CFLAGS+=" -Wno-expansion-to-defined "
```
Comment (riversand963) [inline build_tools/fbcode_config_platform009.sh:144]: Should this be kept or removed?

Comment (siying) [reply build_tools/fbcode_config_platform009.sh:144]: Good catch... Let me revert it.

**Comment by siying** [comment]
"clang analyze" has passed by the way.

**Pull request closed without merging**